server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/create/CreateSnapshotRequest.java

```diff
@@ -56,17 +60,15 @@
  * <li>must not contain invalid file name characters {@link org.elasticsearch.common.Strings#INVALID_FILENAME_CHARS} </li>
  * </ul>
  */
-public class CreateSnapshotRequest extends MasterNodeRequest<CreateSnapshotRequest> implements IndicesRequest {
+public class CreateSnapshotRequest extends MasterNodeRequest<CreateSnapshotRequest> {
 
     private String snapshot;
 
     private String repository;
 
-    private String[] indices = EMPTY_ARRAY;
-
-    private String[] templates = EMPTY_ARRAY;
+    private List<RelationName> relationNames = new ArrayList<>();
```

Comment:
personal preference, no need to change if you don't agree:
Since we don't add relation/partitionNames to other methods than the `in` constructor, I'd prefer to initialize with `List.of()` here, and for `partitionNames` initialize an `ArrayList` in the ctor. `relationNames` are anyway always constructed through the HashSet.